src/FileSystem/AndroidFileSystem.cpp

```diff
@@ -50,6 +55,26 @@ std::filesystem::path AndroidFileSystem::FindPath(const std::filesystem::path& p
 	return path;
 }
 
+bool AndroidFileSystem::IsPathValid(const std::filesystem::path& path)
+{
+	if (path.empty())
+	{
+		return false;
+	}
+
+	if (!Exists(path))
+	{
+		return false;
+	}
+
+	if (!std::filesystem::is_directory(path))
+	{
+		return false;
+	}
+
+	return true;
+}
+
```

Comment:
This won't work. It needs to use the java api

Reply:
@bwrsandman Ok, on it. How come it passed the tests?

Reply:
You will have to setup android studio

Reply:
> @bwrsandman Ok, on it. How come it passed the tests?

It passes the tests because we don't have runtime test on android only compilation test so what you wrote is correct c++ but due to Android ~PITA~ sandboxed filesystem using std::filesystem for android will never work when running the code you need to using android Java API.

This means anything you need the filesystem to do needs to be in a java method, preferably a static one inside the FileSystemInterop class we already have this class call the android API to do what you need in the filesystem. Then in the c++ side you need to use the JNI (a context is already provided to you inside the AndroidFileSystem class) to call this method from the JVM an use it's result in the app.